File: app/src/main/java/de/westnordost/streetcomplete/quests/railway_crossing/AddRailwayCrossingBarrierForm.java
package de.westnordost.streetcomplete.quests.railway_crossing;

import de.westnordost.streetcomplete.R;
import de.westnordost.streetcomplete.quests.ImageListQuestAnswerFragment;
import de.westnordost.streetcomplete.view.Item;

public class AddRailwayCrossingBarrierForm extends ImageListQuestAnswerFragment
{
	@Override protected Item[] getItems() { return new Item[]
		{
			new Item("no", R.drawable.ic_railway_crossing_none, R.string.quest_railway_crossing_barrier_none),
			new Item("half", getCountryInfo().isLeftHandTraffic() ?
				R.drawable.ic_railway_crossing_half_l : R.drawable.ic_railway_crossing_half),
			new Item("double_half", R.drawable.ic_railway_crossing_double_half),
			new Item("full", getCountryInfo().isLeftHandTraffic() ?
				R.drawable.ic_railway_crossing_full_l : R.drawable.ic_railway_crossing_full),
		};
	}
	@Override protected int getItemsPerRow() { return 4; }
}

File: app/src/main/java/de/westnordost/streetcomplete/quests/railway_crossing/AddRailwayCrossingBarrierForm.kt
package de.westnordost.streetcomplete.quests.railway_crossing

import de.westnordost.streetcomplete.quests.AImageListQuestForm
import de.westnordost.streetcomplete.quests.railway_crossing.RailwayCrossingBarrier.CHICANE
import de.westnordost.streetcomplete.quests.railway_crossing.RailwayCrossingBarrier.DOUBLE_HALF
import de.westnordost.streetcomplete.quests.railway_crossing.RailwayCrossingBarrier.FULL
import de.westnordost.streetcomplete.quests.railway_crossing.RailwayCrossingBarrier.GATE
import de.westnordost.streetcomplete.quests.railway_crossing.RailwayCrossingBarrier.HALF
import de.westnordost.streetcomplete.quests.railway_crossing.RailwayCrossingBarrier.NO
import de.westnordost.streetcomplete.view.image_select.DisplayItem

class AddRailwayCrossingBarrierForm : AImageListQuestForm<RailwayCrossingBarrier, RailwayCrossingBarrier>() {

    override val items: List<DisplayItem<RailwayCrossingBarrier>> get() {
        val isPedestrian = element.tags["railway"] == "crossing"
        val answers = if (isPedestrian) listOf(NO, CHICANE, GATE, FULL) else listOf(NO, HALF, DOUBLE_HALF, FULL)
        return answers.toItems(countryInfo.isLeftHandTraffic)
    }

    override val itemsPerRow = 4

    override fun onClickOk(selectedItems: List<RailwayCrossingBarrier>) {
        applyAnswer(selectedItems.single())
    }
}
